src/v/iceberg/rest_client/catalog_client.h

```diff
@@ -118,7 +108,7 @@ class catalog_client {
       http::request_builder request_builder,
       std::optional<iobuf> payload = std::nullopt);
 
-    std::reference_wrapper<client_source> _client_source;
+    std::unique_ptr<http::abstract_client> _http_client;
```

Comment:
shouldn't we still call _http_client->shutdown_and_stop() on cleanup or am missing something?